src/lib/loader/NodeJs.coffee

```diff
@@ -0,0 +1,183 @@
+path = require 'path'
+fs = require 'fs'
+manifest = require 'fbp-manifest'
+{_} = require 'underscore'
+utils = require '../Utils'
+nofloGraph = require '../Graph'
+
+# We allow components to be un-compiled CoffeeScript
+CoffeeScript = require 'coffee-script'
+if typeof CoffeeScript.register != 'undefined'
+  CoffeeScript.register()
+
+registerModules = (loader, modules, callback) ->
+  compatible = modules.filter (m) -> m.runtime in ['noflo', 'noflo-nodejs']
+  componentLoaders = []
+  for m in compatible
+    loader.setLibraryIcon m.name, m.icon if m.icon
+
+    if m.noflo?.loader
+      loaderPath = path.resolve loader.baseDir, m.base, m.noflo.loader
+      componentLoaders.push loaderPath
+
+    for c in m.components
+      loader.registerComponent m.name, c.name, path.resolve loader.baseDir, c.path
+
+  return callback null unless componentLoaders.length
+  done = _.after componentLoaders.length, callback
+  componentLoaders.forEach (loaderPath) =>
+    cLoader = require loaderPath
+    loader.registerLoader cLoader, (err) ->
+      return callback err if err
+      done null
+
+manifestLoader =
+  writeCache: (loader, options, manifest, callback) ->
```

Comment:
And of course the nice thing is that the manifest file can just as well be generated using `fbp-manifest-deps` or other similar tools